### PR TITLE
Add LRO support for NodeJS test gen

### PR DESCRIPTION
**Pull request by shinfan**
Manually tested against Speech API.
Fixes https://github.com/googleapis/toolkit/issues/925
cc @bjwatson 

After this PR I will refresh the tests in the `api-staging-repo`.

### CODE REVIEW

**Comment by shinfan** [comment]
I did end up mocking the whole Operation object and not using gax.operation at all. I believe it is the right way to go since it simplifies the test data setup and we already have the test coverage for the gax stuff from the gax tests.

File: src/main/resources/com/google/api/codegen/nodejs/test.snip [hunk 1]
```diff
@@ -83,6 +83,28 @@
...
+            return new Promise(function(resolve, reject) {
+              if (error) {
+                reject(error)
+              } else {
+                resolve([response]);
+              }
+            });
+          }
+        };
+        return new Promise(function(resolve, reject) {
```
Comment (landrito) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: `reject` is an unneeded parameter.

File: src/main/resources/com/google/api/codegen/nodejs/test.snip [hunk 2]
```diff
@@ -138,16 +160,58 @@
...
+      }).catch(function(err) {
+        done(err);
+      });
+    });
+
+    it('invokes {@test.clientMethodName} with error', function(done) {
+      {@commonInit(test, moduleName)}
+
+      // Mock response
+      {@initCodeLines(test.mockResponse.initCode)}
```
Comment (landrito) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: The response doesn't need to be mocked for an error.

File: src/main/resources/com/google/api/codegen/nodejs/test.snip (hunk 1 above)
Comment (shinfan) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: Done

File: src/main/resources/com/google/api/codegen/nodejs/test.snip (hunk 2 above)
Comment (shinfan) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: Done.

File: src/main/resources/com/google/api/codegen/nodejs/test.snip [hunk 3]
```diff
@@ -99,7 +99,7 @@
             });
           }
         };
-        return new Promise(function(resolve, reject) {
+        return new Promise(function(resolve) {
           resolve([mockOperation]);
         });
       };
```
Comment (landrito) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: nit: `return Promise.resolve([mockOperation]);`

**Comment by shinfan** [comment]
PTAL @landrito

File: src/main/resources/com/google/api/codegen/nodejs/test.snip (hunk 3 above)
Comment (shinfan) [inline src/main/resources/com/google/api/codegen/nodejs/test.snip]: Done

**Pull request merged**